code/modules/client/preferences_savefile.dm

```diff
@@ -90,6 +90,41 @@ SAVEFILE UPDATING/VERSIONING - 'Simplified', or rather, more coder-friendly ~Car
 /datum/preferences/proc/update_character(current_version, savefile/S)
 	return
 
+/// checks through keybindings for outdated unbound keys and updates them
+/datum/preferences/proc/check_keybindings()
+	if(!parent)
+		return
+	var/list/user_binds = list()
+	for (var/key in key_bindings)
+		for(var/kb_name in key_bindings[key])
+			user_binds[kb_name] += list(key)
+	var/list/notadded = list()
+	for (var/name in GLOB.keybindings_by_name)
+		var/datum/keybinding/kb = GLOB.keybindings_by_name[name]
+		if(!length(user_binds[kb.name])) // key is not unbound and not bound to anything
+			var/addedbind = FALSE
+			if(hotkeys)
+				for(var/hotkeytobind in kb.classic_keys)
+					if(!length(key_bindings[hotkeytobind]))
+						LAZYADD(key_bindings[hotkeytobind], kb.name)
+						addedbind = TRUE
+			else
+				for(var/classickeytobind in kb.classic_keys)
+					if(!length(key_bindings[classickeytobind]))
+						LAZYADD(key_bindings[classickeytobind], kb.name)
+						addedbind = TRUE
+			if(!addedbind)
+				notadded += kb
+	if(length(notadded))
+		to_chat(parent, "<span class='boldannounce'> KEYBINDING CONFLICT!!! </span>")
+		to_chat(parent, "<span class='userdanger'>There are new keybindings that have defaults bound to keys you already set, They will default to Unbound. You can bind them in Setup Character or Game Preferences</span>")
```

Comment:
1. I want to see how obvious this is in the chat, especially with all the clutter caused by subsystem initializes when a server just starts to reboot.

2. I'd like for there to be a way, probably a link, to easily get to the preferences menu from here.

My original suggestion was a popup since that's much harder to ignore (it'd be infrequent enough that people would hopefully not just immediately close it out).

Reply:
for 1.
![image](https://user-images.githubusercontent.com/30060146/94483706-41349d00-01a9-11eb-8d2c-831528b6af62.png)

I'll work on 2. later

Reply:
That seems obvious enough I suppose.

Reply:
2. is now done

Reply:
![image](https://user-images.githubusercontent.com/30060146/94514747-fb4ef780-01ef-11eb-8087-6a9c3724bf41.png)